projects/packages/jetpack-mu-wpcom/src/features/100-year-plan/locked-mode.php

```diff
@@ -0,0 +1,196 @@
+			'default'           => false,
+			'sanitize_callback' => function ( $value ) {
+				return (bool) $value;
+			},
+		)
+	);
+}
+add_action( 'admin_init', 'wpcom_lm_register_settings' );
+add_action( 'rest_api_init', 'wpcom_lm_register_settings' );
+
+/**
+ * Adds settings to the v1 API site settings endpoint.
+ *
+ * @param array $settings A single site's settings.
+ * @return array
+ */
+function wpcom_lm_get_options_v1_api( $settings ) {
+	if ( wpcom_site_has_feature( WPCOM_Features::LOCKED_MODE ) ) {
+		$settings['wpcom_locked_mode'] = (bool) get_option( 'wpcom_locked_mode' );
+	}
+
+	return $settings;
+}
+add_filter( 'site_settings_endpoint_get', 'wpcom_lm_get_options_v1_api' );
+
+/**
+ * Updates settings via public-api.wordpress.com.
+ *
+ * @param array $input             Associative array of site settings to be updated.
+ *                                 Cast and filtered based on documentation.
+ * @param array $unfiltered_input  Associative array of site settings to be updated.
+ *                                 Neither cast nor filtered. Contains raw input.
+ * @return array
+ */
+function wpcom_lm_update_options_v1_api( $input, $unfiltered_input ) {
+	if ( isset( $unfiltered_input['wpcom_locked_mode'] ) ) {
+		$input['wpcom_locked_mode'] = (bool) $unfiltered_input['wpcom_locked_mode'];
+	}
+
+	return $input;
+}
+add_filter( 'rest_api_update_site_settings', 'wpcom_lm_update_options_v1_api', 10, 2 );
+
+/**
+ * Registers the settings section and fields.
+ */
+function wpcom_lm_settings_page_init() {
+	if ( ! wpcom_site_has_feature( WPCOM_Features::LOCKED_MODE ) ) {
+		return;
+	}
+
+	add_settings_section(
+		'wpcom_enhanced_ownership_section',
+		__( 'Enhanced Ownership', 'jetpack-mu-wpcom' ),
+		'', // No callback needed.
+		'general'
+	);
+
+	add_settings_field(
+		'wpcom_locked_mode',
+		__( 'Locked Mode', 'jetpack-mu-wpcom' ),
+		'wpcom_locked_mode_render',
+		'general',
+		'wpcom_enhanced_ownership_section',
+		array(
+			'label_for' => 'wpcom_locked_mode',
+		)
+	);
+}
+add_action( 'admin_init', 'wpcom_lm_settings_page_init' );
+
+/**
+ * Renders the Locked Mode settings markup.
+ */
+function wpcom_locked_mode_render() {
+	?>
+	<input type="checkbox" id="wpcom_locked_mode" name="wpcom_locked_mode" <?php checked( get_option( 'wpcom_locked_mode' ) ); ?>>
+	<label for="wpcom_locked_mode"><?php esc_html_e( 'Enable Locked Mode', 'jetpack-mu-wpcom' ); ?></label>
+	<p class="description"><?php esc_html_e( 'Prevents new post and page from being created as well as existing posts and pages from being edited, and closes comments site wide.', 'jetpack-mu-wpcom' ); ?></p>
+	<?php
+}
+
+/**
+ * Removes post and page creation/editing capabilities for locked mode sites.
+ *
+ * @param array  $caps The user's capabilities.
+ * @param string $cap  The capability name.
+ * @return array
+ */
+function wpcom_lm_remove_post_capabilities( $caps, $cap ) {
+	// Return if option is not set.
+	if ( ! get_option( 'wpcom_locked_mode' ) ) {
```

Comment:
Not sure how smart it is to add an option check to every cap check. I wonder if we could move it to an init callback and only add this map_meta_cap filter when needed.

Reply:
That is a good idea. To be on the safe side.